services/slack/config.go

```diff
@@ -52,3 +57,37 @@ func (c Config) Validate() error {
 	}
 	return nil
 }
+
```

Comment:
The NewConfig() method is now not sufficient to set defaults.

 Because its possible that there are multiple config structs created, the toml parser will create a new one using reflection which will not call the NewConfig method.

The config package we use checks for config structs that implement an `ApplyDefaults` method and calls it on newly created config structs.

For example:

```
func (c *Config) ApplyDefaults() {
     if c.Username == "" {
        c.Username = DefaultUsername
    } 
}
```

It would be best to use the ApplyDefaults method within the NewConfig method to ensure they are the same.